ee.ria.qdigidoc4.yml

```diff
@@ -0,0 +1,212 @@
+app-id: ee.ria.qdigidoc4
+command: run.sh
+runtime: org.kde.Platform
+runtime-version: '6.6'
+sdk: org.kde.Sdk
+rename-icon: qdigidoc4
+
+finish-args:
+  # libdigidocpp stores tsl files here
+  - --persist=.digidocpp
+  # Network is needed for some parts of the application's functionality
+  - --share=network
+  # Wayland
+  - --socket=wayland
+  # X11
+  - --socket=fallback-x11
+  - --share=ipc
+  # Smartcard
+  - --socket=pcsc
+  - --device=all
+
+cleanup:
+  - /include
+  - /lib/cmake
+  - /lib/pkgconfig
+  - /etc/bash_completion.d
+  - /share/doc
+  - /share/man
+
+modules:
+  - name: flatbuffers
+    buildsystem: cmake-ninja
+    sources:
+      - type: archive
+        url: https://github.com/google/flatbuffers/archive/refs/tags/v24.3.25.zip
+        sha256: e706f5eb6ca8f78e237bf3f7eccffa1c5ec9a96d3c1c938f08dc09aab1884528
+  - name: xerces-c
+    buildsystem: cmake-ninja
+    sources:
+      - type: archive
+        url: https://github.com/apache/xerces-c/archive/refs/tags/v3.2.4.tar.gz
+        sha256: 8dfaa30d6a641bda113625ef65e43c433e8ffd94fadd3b8d39dfe6faf450f26d
+
+  - name: xalan-c
+    buildsystem: cmake-ninja
+    sources:
+      - type: archive
+        url: https://github.com/apache/xalan-c/releases/download/Xalan-C_1_12_0/xalan_c-1.12.tar.gz
+        sha256: ee7d4b0b08c5676f5e586c7154d94a5b32b299ac3cbb946e24c4375a25552da7
+      - type: patch
+        path: xalan-c/CMakeLists.patch
+  
+  # We use prebuilt xsd since its build seems to take a lot of memory and time. Git history has a
+  # non-prebuilt version of the xsd module if needed.
+  - name: xsd
+    sources:
+      - type: archive
+        url: https://www.codesynthesis.com/download/xsd/4.0/linux-gnu/x86_64/xsd-4.0.0-x86_64-linux-gnu.tar.bz2
```

Comment:
how will this work on aarch64? You should build from source.

Reply:
xsd had a large gap in releases between 2014 and 2023. 4.0.0 is the last version that uses make, but does not build due to changes in C++17. Recent versions use build2 which I don't have a clue how to use with flatpak, if that's even possible.

If we don't target aarch64, would it be acceptable to use the binary?

Reply:
> but does not build due to changes in C++17.

you can tell gcc to not use C++17.